dspace_rest_client/client.py

```diff
@@ -1163,3 +1186,151 @@ def update_resource_policy_group(self, policy_id, group_uuid):
+            data = parse_json(response)
+            allowances = data.get('_embedded', {}).get('clarinlruallowances')
+            if allowances:
+                _logger.info(f"Fetched {len(allowances)} CLARIN LRU allowances.")
+                return allowances
+            _logger.warning("No CLARIN LRU allowances found.")
+        except Exception as e:
+            _logger.error(f"Error fetching CLARIN LRU allowances [{url}]: {e}")
+        return None
+
+    def get_clarinlruallowances_by_bitstream_and_user(self, bitstream_uuid, user_uuid):
+        """
+        Fetch user allowances for a specific bitstream and user.
+        """
+        url = f'{self.API_ENDPOINT}/core/clarinlruallowance/search/byBitstreamAndUser'
+        params = {'bitstreamUUID': bitstream_uuid, 'userUUID': user_uuid}
+        try:
+            response = self.api_get(url, params=params)
+            data = parse_json(response)
+            allowances = data.get('_embedded', {}).get('clarinlruallowances')
+            if allowances:
+                _logger.info(f"Found {len(allowances)} user allowance(s).")
+                return allowances
+            _logger.warning(f"No user allowances found for user: {user_uuid} and bitstream: {bitstream_uuid}")
+        except Exception as e:
+            _logger.error(f"Error fetching user allowances: {e}")
+        return None
+
+
+    def create_clarinlruallowances(self, bitstream_uuid):
+        """
+        Create clarinlruallowances for a bitstream for logged user
+        by managing user metadata of bitstream.
+        """
+        url = f'{self.API_ENDPOINT}/core/clarinusermetadata/manage'
+        params = {'bitstreamUUID': bitstream_uuid}
+        metadata_payload = [
+            {"metadataKey": "NAME", "metadataValue": "Test"}
+        ]
+        try:
+            response = self.api_post(url, json=metadata_payload, params=params)
+            if response.status_code == 200:
+                _logger.info(f"User metadata access managed for bitstream: {bitstream_uuid}")
+                return True
+            _logger.warning(f"Failed to manage user metadata: {response.status_code}")
+        except Exception as e:
+            _logger.error(f"Error managing user metadata: {e}")
+        return False
+
+
+    def logout(self):
+        """
+        Log out from the DSpace session.
+        """
+        try:
+            response = self.session.post(f'{self.API_ENDPOINT}/authn/logout', headers=self.request_headers)
+            if response.status_code == 204:
+                self.session.cookies.clear()
+                self.session.headers.pop('Authorization', None)
+                _logger.info("Logout successful.")
+                return True
+            else:
+                _logger.error(f"Logout failed: {response.status_code} - {response.text}")
+        except Exception as e:
+            _logger.error(f"Logout error: {e}")
+        return False
+
+    def get_http_status(self, url):
```

Comment:
why is this here?

Reply:
I use it to test whether the provided URL is accessible by the current user.